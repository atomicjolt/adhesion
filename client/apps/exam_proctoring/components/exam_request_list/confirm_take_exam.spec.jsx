import React from 'react';
import { shallow } from 'enzyme';
import ConfirmExam from './confirm_take_exam';

describe('Confirm Take Exam', () => {
  let result;
  let props;
  let status;

  beforeEach(() => {
    status = null;
    props = {
      takeExam: () => { status = 'taking'; },
      closeModal: () => { status = 'closing'; },
    };
    result = shallow(<ConfirmExam {...props} />);
  });

  it('matches the snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('cancel button cancels', () => {
    const close = result.find('.qa-cancel-take-exam');
    close.simulate('click');
    expect(status).toBe('closing');
  });

  it('start button starts quiz', () => {
    const start = result.find('.qa-begin-take-exam');
    start.simulate('click');
    expect(status).toBe('taking');
  });
});
