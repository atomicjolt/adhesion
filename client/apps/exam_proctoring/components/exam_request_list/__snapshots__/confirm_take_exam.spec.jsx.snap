// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Take Exam matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <confirmTakeExam
    closeModal={[Function]}
    takeExam={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Are you sure?
        </h2>,
        <div
          style={
            Object {
              "marginBottom": "20px",
            }
          }
        >
          You are about to begin a quiz on the behalf of a student. This action will save the attempt and scores of the quiz attempt to the students account. You will be recorded as the proctor who entered the answers. Time limits, shuffled questions, and other factors will still apply. Only select Begin Quiz if you are prepared to enter answers for this student.
        </div>,
        <button
          className="qa-cancel-take-exam"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#ededed",
              "border": "none",
              "cursor": "pointer",
              "padding": "10px 20px",
            }
          }
        >
          Cancel
        </button>,
        <button
          className="qa-begin-take-exam"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#c1a17a",
              "border": "none",
              "color": "white",
              "cursor": "pointer",
              "float": "right",
              "padding": "10px 20px",
            }
          }
        >
          Begin Quiz
        </button>,
      ],
      "style": Object {
        "backgroundColor": "white",
        "borderRadius": "5px",
        "boxShadow": "0px 0px 5px #555555",
        "left": "20vw",
        "padding": "20px",
        "position": "fixed",
        "right": "20vw",
        "top": "10vh",
        "zIndex": "2",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Are you sure?",
        },
        "ref": null,
        "rendered": "Are you sure?",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You are about to begin a quiz on the behalf of a student. This action will save the attempt and scores of the quiz attempt to the students account. You will be recorded as the proctor who entered the answers. Time limits, shuffled questions, and other factors will still apply. Only select Begin Quiz if you are prepared to enter answers for this student.",
          "style": Object {
            "marginBottom": "20px",
          },
        },
        "ref": null,
        "rendered": "You are about to begin a quiz on the behalf of a student. This action will save the attempt and scores of the quiz attempt to the students account. You will be recorded as the proctor who entered the answers. Time limits, shuffled questions, and other factors will still apply. Only select Begin Quiz if you are prepared to enter answers for this student.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cancel",
          "className": "qa-cancel-take-exam",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#ededed",
            "border": "none",
            "cursor": "pointer",
            "padding": "10px 20px",
          },
        },
        "ref": null,
        "rendered": "Cancel",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Begin Quiz",
          "className": "qa-begin-take-exam",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#c1a17a",
            "border": "none",
            "color": "white",
            "cursor": "pointer",
            "float": "right",
            "padding": "10px 20px",
          },
        },
        "ref": null,
        "rendered": "Begin Quiz",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Are you sure?
          </h2>,
          <div
            style={
              Object {
                "marginBottom": "20px",
              }
            }
          >
            You are about to begin a quiz on the behalf of a student. This action will save the attempt and scores of the quiz attempt to the students account. You will be recorded as the proctor who entered the answers. Time limits, shuffled questions, and other factors will still apply. Only select Begin Quiz if you are prepared to enter answers for this student.
          </div>,
          <button
            className="qa-cancel-take-exam"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#ededed",
                "border": "none",
                "cursor": "pointer",
                "padding": "10px 20px",
              }
            }
          >
            Cancel
          </button>,
          <button
            className="qa-begin-take-exam"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#c1a17a",
                "border": "none",
                "color": "white",
                "cursor": "pointer",
                "float": "right",
                "padding": "10px 20px",
              }
            }
          >
            Begin Quiz
          </button>,
        ],
        "style": Object {
          "backgroundColor": "white",
          "borderRadius": "5px",
          "boxShadow": "0px 0px 5px #555555",
          "left": "20vw",
          "padding": "20px",
          "position": "fixed",
          "right": "20vw",
          "top": "10vh",
          "zIndex": "2",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Are you sure?",
          },
          "ref": null,
          "rendered": "Are you sure?",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You are about to begin a quiz on the behalf of a student. This action will save the attempt and scores of the quiz attempt to the students account. You will be recorded as the proctor who entered the answers. Time limits, shuffled questions, and other factors will still apply. Only select Begin Quiz if you are prepared to enter answers for this student.",
            "style": Object {
              "marginBottom": "20px",
            },
          },
          "ref": null,
          "rendered": "You are about to begin a quiz on the behalf of a student. This action will save the attempt and scores of the quiz attempt to the students account. You will be recorded as the proctor who entered the answers. Time limits, shuffled questions, and other factors will still apply. Only select Begin Quiz if you are prepared to enter answers for this student.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
            "className": "qa-cancel-take-exam",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#ededed",
              "border": "none",
              "cursor": "pointer",
              "padding": "10px 20px",
            },
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Begin Quiz",
            "className": "qa-begin-take-exam",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#c1a17a",
              "border": "none",
              "color": "white",
              "cursor": "pointer",
              "float": "right",
              "padding": "10px 20px",
            },
          },
          "ref": null,
          "rendered": "Begin Quiz",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
